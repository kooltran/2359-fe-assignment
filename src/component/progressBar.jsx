import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { ProgressBar } from "react-bootstrap";

export class ProgressBarCmp extends React.Component {
  static propTypes = {
    currentProgress: PropTypes.object.isRequired,
    bars: PropTypes.array.isRequired,
    limit: PropTypes.number.isRequired
  };

  render() {
    const { bars, limit, currentProgress } = this.props;
    return bars.map((bar, idx) => {
      const currentProgressBarVal = currentProgress[`${idx + 1}`];
      const barValue =
        currentProgressBarVal !== undefined ? currentProgressBarVal : bar;
      return (
        <div key={idx} className="progress-wrapper">
          <ProgressBar
            className="progress-item"
            now={barValue}
            max={limit}
            variant={barValue >= limit ? "danger" : "info"}
            label={`${barValue}%`}
          />
        </div>
      );
    });
  }
}

const mapStateToProps = state => {
  return {
    currentProgress: state.updateProgressBarReducer.currentProgress
  };
};

export default connect(mapStateToProps, null)(ProgressBarCmp);
