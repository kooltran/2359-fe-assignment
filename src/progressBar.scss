.progress {
  &-item {
    height: 50px;
  }

  &-wrapper {
    margin-bottom: 20px;
  }

  &-dropdown {
    margin-bottom: 30px;
    width: 300px;
  }

  &-bar {
    font-size: 18px;
    font-weight: 600;
  }

  &-loading {
    text-align: center;
  }

  &-button {
    width: 80px;
    height: 50px;
    margin-right: 10px;
    background: #17aa56;
    color: #fff;
    outline: none;
    box-shadow: 0 5px #119e4d;
    border: 0;
    border-radius: 5px;

    &:focus {
      outline: none;
    }
  }
}
