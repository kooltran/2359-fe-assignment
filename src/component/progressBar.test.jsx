import React from "react";
import { shallow } from "enzyme";
import { ProgressBarCmp } from "./progressBar";

describe("renders ProgressBar component", () => {
  const baseProps = {
    currentProgress: {},
    bars: [10, 20, 30],
    limit: 10
  };

  it("should render properly with base props", () => {
    expect(shallow(<ProgressBarCmp {...baseProps} />)).toMatchSnapshot();
  });
});
