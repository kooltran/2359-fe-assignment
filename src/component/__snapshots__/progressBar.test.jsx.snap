// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProgressBar component should render properly with base props 1`] = `
Array [
  <div
    className="progress-wrapper"
    key="0"
  >
    <ProgressBar
      animated={false}
      className="progress-item"
      isChild={false}
      label="10%"
      max={10}
      min={0}
      now={10}
      srOnly={false}
      striped={false}
      variant="danger"
    />
  </div>,
  <div
    className="progress-wrapper"
    key="1"
  >
    <ProgressBar
      animated={false}
      className="progress-item"
      isChild={false}
      label="20%"
      max={10}
      min={0}
      now={20}
      srOnly={false}
      striped={false}
      variant="danger"
    />
  </div>,
  <div
    className="progress-wrapper"
    key="2"
  >
    <ProgressBar
      animated={false}
      className="progress-item"
      isChild={false}
      label="30%"
      max={10}
      min={0}
      now={30}
      srOnly={false}
      striped={false}
      variant="danger"
    />
  </div>,
]
`;
